### ISSUE TITLE
`lexer%chunk` out of bounds in Windows build

### ISSUE THREAD

**Issue by perazz** [toml-f/jonquil#15]
Hi, 

while building an [fpm PR](https://github.com/fortran-lang/fpm/pull/879) that uses Jonquil, the Windows build fails at the first JSON serialization test with the following message: 

```
# Starting serialize-git-target ... (4/24)
At line 184 of file build\dependencies\jonquil\src\jonquil\lexer.f90
Fortran runtime error: Substring out of bounds: upper bound (132) of 'lexer' exceeds string length (131)

Error termination. Backtrace:
Could not print backtrace: libbacktrace could not find executable to open
```

I am not sure if the problem is upstream, but in the same routine, an out-of-bounds error is possible: 

```fortran

   ! If lexer is exhausted, return EOF as early as possible
   if (pos > len(lexer%chunk)) then
      token = toml_token(token_kind%eof, prev, pos)
      return
   end if

  ! [fp] at this point, it is possible that pos == len(lexer%chunk), which would cause the out of bounds

   select case(lexer%chunk(pos:pos))
   case(" ", toml_escape%tabulator, toml_escape%newline, toml_escape%carriage_return)
      do while(any(lexer%chunk(pos+1:pos+1) == [" ", toml_escape%tabulator, &
            & toml_escape%newline, toml_escape%carriage_return]) .and. pos < len(lexer%chunk))  ! [fp] here
         pos = pos + 1
      end do
      token = toml_token(token_kind%whitespace, prev, pos)
      return
```

Possible (untested) solution: 

```fortran
     if (pos<len(lexer%chunk)) then  
     do while(any(lexer%chunk(pos+1:pos+1) == [" ", toml_escape%tabulator, &
            & toml_escape%newline, toml_escape%carriage_return]) .and. pos < len(lexer%chunk))  ! [fp] here
         pos = pos + 1
     end do
     endif
```

**Comment by perazz** [comment]
The ASCII codes of the test string are: 
input string:  123 34 100 101 115 99 114 105 112 116 111 114 34 58 32 34 114 101 118 105 115 105 111 110 34 44 34 117 114 108 34 58 32 34 104 116 116 112 115 58 47 47 103 105 116 104 117 98 46 99 111 109 47 117 114 98 97 110 106 111 115 116 47 77 95 67 76 73 50 46 103 105 116 34 44 34 111 98 106 101 99 116 34 58 32 34 55 50 54 52 56 55 56 99 100 98 49 98 97 102 102 55 51 50 51 99 99 52 56 53 57 54 100 56 50 57 99 99 102 101 55 55 53 49 98 56 34 125 10 10 10

This fails with the fpm CI gfortran: 
```
GNU Fortran (MinGW-W64 x86_64-posix-seh, built by Brecht Sanders) 9.4.0
```

But not with the current config from MSYS2: 
```
GNU Fortran (Rev10, Built by MSYS2 project) 12.2.0
```

I believe we may have hit a bug with gfortran 9

**Comment by perazz** [comment]
Solved by #17
